backends-common/postgres/src/test/java/org/apache/james/backends/postgres/PostgresExecutorThreadSafetyTest.java

```diff
@@ -51,7 +51,7 @@ class PostgresExecutorThreadSafetyTest {
     @RegisterExtension
     static PostgresExtension postgresExtension = PostgresExtension.empty();
 
-    private static PostgresExecutor postgresExecutor;
+    private static DefaultPostgresExecutor postgresExecutor;
```

Comment:
Just use the same Pg executor everywhere please?

Reply:
We can not use pool Executor for postgres-mailbox (that was used RLS)
We can use pool for userDao, domainDao (noRLS DAO)... but not everywhere

Reply:
I would prefer having one and only one central PostgresExecutor.

What I propose:

 - Introduce a PgExecutor interface, used everywhere.
 - Modify DefaultPGExecutor so that:
       - We use **ONE pool** for all non RLS restricted query
       - We use one connection for per domain for RLS
       
(IE combine the two behaviour in one class)

Then we (later) could experiment various behaviours based on the needs. Eg: one connection pool per RLS.

What do you think?